.evergreen/verify-packaged-artifact.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+set -e
+set -x
+
+# Use tmp directory for all gpg operations/the rpm database
+GPG_HOME=$(mktemp -d)
+TMP_FILE=$(mktemp)
+MONGOSH_KEY="https://pgp.mongodb.com/mongosh.asc"
+ARTIFACTS_DIR="dist"
+
+trap_handler() {
+  local code=$?
+  if [ $code -eq 0 ]; then
+    echo "Verification successful"
+  else
+    echo "Verification failed with exit code $code"
+    cat "$TMP_FILE"
+  fi
+  rm -f "$TMP_FILE"
+  rm -rf "$GPG_HOME"
+  exit $code
+}
+
+trap trap_handler ERR EXIT
+
+verify_using_gpg() {
+  echo "Verifying $1 using gpg"
+  gpg --homedir $GPG_HOME --verify $ARTIFACTS_DIR/$1.sig $ARTIFACTS_DIR/$1 > "$TMP_FILE" 2>&1
+}
+
+verify_using_powershell() {
+  echo "Verifying $1 using powershell"
+  powershell Get-AuthenticodeSignature -FilePath $ARTIFACTS_DIR/$1 > "$TMP_FILE" 2>&1
+}
+
+verify_using_codesign() {
+  echo "Verifying $1 using codesign"
+  codesign -dv --verbose=4 $ARTIFACTS_DIR/$1 > "$TMP_FILE" 2>&1
+}
+
+verify_using_rpm() {
+  # RPM packages are signed using gpg and the signature is embedded in the package.
+  # Here, we need to import the key in `rpm` and then verify the signature.
+  echo "Importing key into rpm"
+  rpm --dbpath "$GPG_HOME" --import $MONGOSH_KEY > "$TMP_FILE" 2>&1
+  # Even if the file is not signed, the command below will exit with 0 and output something like: digests OK
+  # So we need to check the output of the command to see if the file is signed successfully.
+  echo "Verifying $1 using rpm"
+  output=$(rpm --dbpath "$GPG_HOME" -K $ARTIFACTS_DIR/$1)
+
+  # Check if the output contains the string "pgp md5 OK"
```

Comment:
```suggestion
  # Check if the output contains the string "digests signatures OK"
```

although admittedly the entire comment is just very close to describing the literal content of the next line anyway :)